main.go

```diff
@@ -18,5 +18,223 @@
 
 package main
 
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/networkservicemesh/api/pkg/api/registry"
+
+	nested "github.com/antonfisher/nested-logrus-formatter"
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+	"github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig"
+	"github.com/spiffe/go-spiffe/v2/svid/x509svid"
+	"github.com/spiffe/go-spiffe/v2/workloadapi"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+
+	"github.com/networkservicemesh/sdk/pkg/tools/log"
+	"github.com/networkservicemesh/sdk/pkg/tools/log/logruslogger"
+	"github.com/networkservicemesh/sdk/pkg/tools/spiffejwt"
+	"github.com/networkservicemesh/sdk/pkg/tools/token"
+	"github.com/networkservicemesh/sdk/pkg/tools/tracing"
+)
+
+const registryAddr = "registry:5002"
+
+func handleError(w http.ResponseWriter, err error) {
+	fmt.Println("Error: ", err)
+	http.Error(w, "Error encoding JSON", http.StatusInternalServerError)
+}
+
 func main() {
+	ctx, cancel := signal.NotifyContext(
+		context.Background(),
+		os.Interrupt,
+		syscall.SIGHUP,
+		syscall.SIGTERM,
+		syscall.SIGQUIT,
+	)
+	defer cancel()
+
+	// Setup logger
+	log.EnableTracing(true)
+	logrus.SetFormatter(&nested.Formatter{})
+	ctx = log.WithLog(ctx, logruslogger.New(ctx, map[string]interface{}{"cmd": os.Args[:1]}))
+	logger := log.FromContext(ctx)
+
+	// Setup Spire
+	source, err2 := workloadapi.NewX509Source(ctx)
```

Comment:
I think we can reuse `err` as the error name for the most cases

Reply:
done